src/ApiCore/GapicClientTrait.php

```diff
@@ -95,117 +99,193 @@ private static function getGapicVersion(array $options)
-     *           for example usage. Passing a value of null is equivalent to a value of
-     *           ['retriesEnabled' => false]. Retry settings provided in this setting override the
-     *           settings in $clientConfigPath.
-     *     @type callable $authHttpHandler A handler used to deliver PSR-7
-     *           requests specifically for authentication. Should match a
-     *           signature of
-     *           `function (RequestInterface $request, array $options) : ResponseInterface`.
-     *     @type callable $httpHandler A handler used to deliver PSR-7 requests.
-     *           Should match a signature of
-     *           `function (RequestInterface $request, array $options) : PromiseInterface`.
-     *           NOTE: This option is only valid when utilizing the REST transport.
-     *     @type string|TransportInterface $transport
-     *           The transport used for executing network requests. May be either
-     *           the string `rest` or `grpc`. Additionally, it is possible
-     *           to pass in an already instantiated transport. Defaults to `grpc`
-     *           if gRPC support is detected on the system.
-     *     @type string $libName
+     * @type string $libName
      *           The name of the client application.
-     *     @type string $libVersion
+     * @type string $libVersion
      *           The version of the client application.
-     *     @type string $gapicVersion
+     * @type string $gapicVersion
      *           The code generator version of the GAPIC library.
      * }
      * @throws ValidationException
      */
     protected function setClientOptions(array $options)
     {
         $this->validateNotNull($options, [
+            'serviceAddress',
             'serviceName',
             'descriptorsConfigPath',
-            'clientConfigPath'
+            'clientConfig',
+            'disableRetries',
         ]);
-        $transport = isset($options['transport'])
-            ? $options['transport']
-            : null;
-        $clientConfig = json_decode(
-            file_get_contents($options['clientConfigPath']),
-            true
-        );
+        $this->validate($options, [
+            'auth',
+            'authConfig',
+            'transport',
+            'transportConfig',
+        ]);
+
+        $transport = $options['transport'] ?: self::defaultTransport();
+        $transportConfig = $options['transportConfig'] ?: [];
+        $clientConfig = $options['clientConfig'];
+        if (is_string($clientConfig)) {
+            $clientConfig = json_decode(file_get_contents($clientConfig), true);
+        }
         $this->serviceName = $options['serviceName'];
-        $this->retrySettings = RetrySettings::load(
+        $retrySettings = RetrySettings::load(
             $this->serviceName,
             $clientConfig,
-            $this->pluck('retryingOverride', $options, false)
```

Comment:
What if we removed [the retryingOverride setting](https://github.com/googleapis/gax-php/blob/master/src/ApiCore/RetrySettings.php#L261) on `RetrySettings::load()` and replaced it with a bool `disableRetries`? We could pass in the value we get at this point and skip needing to reiterate the list again below.